content/things/rust-to-webassembly.md

```diff
@@ -0,0 +1,637 @@
+---json
+{
+  "title": "Rust to WebAssembly the hard way",
+  "date": "2023-02-08",
+  "socialmediaimage": "social.jpg",
+  "live": false
+}
+
+---
+
+For many, Rust is the first choice when targeting WebAssembly. Let’s take full control.
+
+<!-- more -->
+
+A long old time ago, I wrote a blog post on [how to compile C to WebAssembly without Emscripten][c to wasm], i.e. without the default tool that makes that process easy. In Rust, the tool that makes WebAssembly easy is called [wasm-bindgen], and we are going to ditch it! At the same time, Rust is a bit different in that WebAssembly was always a first-class target and the standard library is laid out to support it out of the box.
+
+## Rust to WebAssembly 101
+
+Let’s see how we can get Rust to emit WebAssembly with as little deviation from the standard Rust workflow as possible. If you look around the internet, a lot of articles and guides tell you to create a Rust library project with `cargo init --lib` and add this line to your `Cargo.toml`:
```

Comment:
[nit] the Internet should [in theory](https://proofed.co.uk/writing-tips/when-should-i-capitalise-internet/) be capitalized